modules/oe_subscriptions_anonymous/tests/src/Functional/SubscribeTest.php

```diff
@@ -78,10 +79,13 @@ public function testForm(): void {
     ]);
 
     $assert_session = $this->assertSession();
-    $this->drupalGet($article->toUrl());
 
-    // Click subscribe link.
+    // Visit the article node, and click subscribe.
+    $this->drupalGet($article->toUrl());
     $this->clickLink('Subscribe to this article');
+
+    // Without javascript, the subscribe form opens in a new page.
+    // With javascript, it would open in a modal.
```

Comment:
My goal here is that you can skim over the test by just reading the comments.

For the best result, I believe it is fine to have some level of redundancy, and not be afraid to state the obvious.